frontend/packages/console-dynamic-plugin-sdk/src/extensions/storage-class-provisioner.ts

```diff
@@ -35,6 +44,10 @@ export type ProvisionerDetails = {
       validationMsg?: string;
     };
   };
+  mutators?: {
+    mutate: CodeRef<(storageClass: StorageClass) => StorageClass>;
+    required?: CodeRef<(storageClass: StorageClass) => boolean> | boolean;
```

Comment:
why do we need separate "required" field ??
mutate itself takes current SC obj and returns updated one... we can choose to update/not update the current SC obj based on some conditions in mutate itself...